### ISSUE TITLE
Text selection handle color should be accent color by default

### ISSUE THREAD

**Issue by robertoscaramuzzi** [flutter/flutter#17635]
The text selection handle color in text widgets is primarySwatch[300] by default (https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/theme_data.dart#L135), but it should be the accent color. 

The relevant guideline appears to be gone from the material documentation, but it's documented Google-internally at: https://material.googleplex.com/patterns/selection.html#selection-item-selection

**Comment by jpc-ae** [comment]
This old issue appears to be wrong according to native testing. According to android and ios text input design specs, the cursor color should default to primary color, and the handles on both android and ios always default to match the cursor color. Of course, these can all be overwritten, but the default should be the primary color to match the outline color of the new text input design specifications.

I would suggest this issue be closed as well now that the material component redesign is being worked on.

**Comment by Deepanshu-Rohilla** [comment]
I think accent color is the better choice. Sometimes, primary color may be same as background color of the input text field.

**Comment by jpc-ae** [comment]
> I think accent color is the better choice. Sometimes, primary color may be same as background color of the input text field.

It could just be overridden in the theme in that case. The default should still be according to material design specs. Your case may be why it's not a rule though and only a default.

**Comment by Deepanshu-Rohilla** [comment]
> > I think accent color is the better choice. Sometimes, primary color may be same as background color of the input text field.
> 
> It could just be overridden in the theme in that case. The default should still be according to material design specs. Your case may be why it's not a rule though and only a default.

After specifying accent color in the themes, it should automatically adapt that color. I have seen this bug in many open sourced flutter applications.
Here are some screenshots for better understanding.

WHAT YOU GET
![WhatsApp Image 2020-06-23 at 9 45 38 PM (5)](https://user-images.githubusercontent.com/52520071/85434153-3fa07b00-b5a3-11ea-953c-f871dcae0ba8.jpeg)

WHAT YOU WANT
![WhatsApp Image 2020-06-23 at 9 45 38 PM (1)](https://user-images.githubusercontent.com/52520071/85434188-4af3a680-b5a3-11ea-9e4d-16dccedbf6da.jpeg)

**Comment by jpc-ae** [comment]
Yeah, except the cursor should be orange too in the second image. I could create a pull request to set those correctly, but I think it would just be a stopgap while the material design is still being worked on. My point here was more that this issue is about changing it to the accent color, when it really should be primary color by default. Would it make sense for me to open the pull request?

**Comment by Deepanshu-Rohilla** [comment]
Well, you are absolutely right. It is not kind of a `bug` that require urgent fix. Rather, it is more of an `enhancement`.

**Comment by Deepanshu-Rohilla** [comment]
I once again ran into similar issue.
I have specified almost every color in my ThemeData. Still I am getting that standard dark theme color of flutter at random places. For example: 
What I am getting


![WhatsApp Image 2020-06-27 at 3 30 12 PM](https://user-images.githubusercontent.com/52520071/85919753-de6ef500-b88b-11ea-8549-fbbd9d62bdd6.jpeg)

What I want


![WhatsApp Image 2020-06-27 at 3 33 43 PM](https://user-images.githubusercontent.com/52520071/85919755-e29b1280-b88b-11ea-9ea0-9ae22f49e396.jpeg)

Basically, if I am specifying all the colors, then that "default" color should not come anywhere.
We can maybe give an option to totally overwrite that color.

**Comment by jpc-ae** [comment]
That's a widget issue rather than a theme issue, unrelated to the text selection handle. The Checkbox widget has a hard-coded color in its build method which needs to be overridden by the checkColor property. The other color may come from the theme's toggleableActiveColor, unselectedWidgetColor, focusColor or hoverColor properties. Basically it would fall under the Material Design Overhall being worked on. I would look at a separate issue for that one.

I will look into creating a pull for all the items listed here though to try to fix some of the errors people are having at least :-) Hopefully it will get accepted.

**Comment by HansMuller** [comment]
@jpc-ae - we'd appreciate such a PR however it's likely to be complicated by the fact that we're in the middle of overhauling the theme system, see [flutter.dev/go/material-theme-system-updates](https://flutter.dev/go/material-theme-system-updates).

**Comment by jpc-ae** [comment]
@HansMuller Yes, thank you. I guess my pr would be mainly for temporary backwards compatibility, but I suppose it would depend on how long the Material Design overhaul take if it  ever makes it into a release (in anything but a deprecated state).

If I wanted to help out with a specific component of the overhaul as well, would that be a waste of time? Or is there a timeline I could check so I could work on those that won't be for a while?

**Comment by jpc-ae** [comment]
Okay, I created a pull request #60821. I'm not sure which test I should update for it though, if any.

**Comment by darrenaustin** [comment]
I believe this is handled by the new `TextSelectionTheme` that we just finished migrating to: #62014, #66061, #66482. The defaults are now based on the primary color for the color scheme, but can be set to anything you want using the new theme.
